tests/test_requests.py

```diff
@@ -240,11 +240,32 @@ def test_request_is_disconnected():
     assert disconnected_after_response
 
 
+def test_request_state_object():
```

Comment:
I think it'd probably be marginally nicer if we tested this by instantiating a request object, and accessing `request.state` on it.